src/framebuffer.rs

```diff
@@ -0,0 +1,123 @@
+use core::ptr;
+use core::slice;
+use header::Tag;
+
+#[derive(Debug, PartialEq)]
+pub struct FramebufferTag<'a> {
+    pub address: u64,
+    pub pitch: u32,
+    pub width: u32,
+    pub height: u32,
+    pub bpp: u8,
+    pub buffer_type: FramebufferType<'a>
+}
+
+#[derive(Debug, PartialEq)]
+pub enum FramebufferType<'a> {
+    Indexed {
+        palette: &'a [FramebufferColor]   
+    },
+    RGB {
+        red: FramebufferField,
+        green: FramebufferField,
+        blue: FramebufferField
+    },
+    Text
+}
+
+#[derive(Debug, PartialEq)]
+pub struct FramebufferField {
+    pub position: u8,
+    pub size: u8
+}
+
+#[derive(Debug, PartialEq)]
+#[repr(C, packed)]
+pub struct FramebufferColor {
+    pub red: u8,
+    pub green: u8,
+    pub blue: u8
+}
+
+struct Reader {
```

Comment:
This seems like a useful utility for other tags.  Can you move it into `src/lib.rs` and make it `pub`?

Reply:
Looks super useful. I wonder if it should be `pub(crate)` tho, as it doesn't seem useful to the user of the library

Reply:
Yes, `pub(crate)` seems like the right choice here